API.pm

```diff
@@ -706,8 +706,20 @@ sub _get {
 
 	if (!$params->{_nocache} && (my $cached = $cache->get($url))) {
 		main::DEBUGLOG && $log->is_debug && $log->debug("found cached response: " . Data::Dump::dump($cached));
-		$cb->($cached);
-		return;
+		my $refresh = 0;
+		if ( $cached->{release_date_stream} ) {
+			my $today = Slim::Utils::DateTime::shortDateF(time, "%Y-%m-%d");
+			if ( $cached->{release_date_stream} ge $today ) {
+				# remove from cache in case previous version of plugin had added it, then set flag to force re-retrieve from Qobuz
+				$refresh = 1;
+				$cache->remove($url);
+			}
+		}
+
+		if ( !$refresh ) {
+			$cb->($cached);
+			return;
+		}
```

Comment:
I'm not sure whether this is read for review already? But instead of adding specific code for all kinds of backwards compatibility issues you could simply bump the cache version number in API/Common.pm (https://github.com/pierrepoulpe/SqueezeboxQobuz/blob/fcc34d6e572af928cf16f61df034d64d0fcadf8c/API/Common.pm#L51): by increasing that number 2 to 3 you'd basically invalidate all of the cache.

I just don't like the idea of starting to add cache cleanup code for all kinds of incompatible changes. The cache should be considered something temporary, which can go away at any time.

Reply:
I can see that this is much cleaner. Might lead to a bit of a spike in Qobuz server access on release day.

Not quite ready for full review/merge yet, Sam has the new version with extra debug code for testing and will commit when he is happy. 

(I have some enhancements to classical playlist handling in my fork, I suppose I should learn how to manage multiple branches rather than attaching code to messages.)